### PR TITLE
Frontend/view cancel appointment

### LINKED ISSUE #164: Appointment tab: View Appointment & Cancel Appointment (Customer)
### **🏁 Setup**
- [x] **Assign** yourself to this issue 

### **✅ To do:**
- [ ] Read [Detailed Use Case: Cancel Appointment](https://github.com/McGill-ECSE321-Winter2021/project-group-09/wiki/Detailed-Use-Cases#use-case-cancel-appointment)
- [ ] **Draw** page mockup using [Whimsical](https://whimsical.com/repairshop-6v7GHoRVFfSSvyvUdeHaQ8) 
(After clicking on the link, you must sign in with your Google Account)

**Frontend**

- [ ] View all the customer's appointments 
- [ ] Customer selects an appointment and deletes it 
- [ ] Confirmation page
- [ ] Implement frontend

### **🆘 If you need help for the issue you’re working on:**

**1) Write** comments in this issue 
**2) Add** label image
**3) Copy** the issue link and **paste** it to Discord's `#🐛issues` channel
**4)** When solved, **strike** the message in `#🐛issues`, **mark it** as `(SOLVED)`  + **remove** label `help wanted` + **write** comments in this issue 

### *Whenever something is done write a comment below.*

### PR DESCRIPTION
**Pull request by JDziewonski98**
image
Viewing + cancelling appointments works, it disables the button based on date, todo is cancel confirmation prompt

### CODE REVIEW

File: RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/controller/AppointmentController.java [hunk 1]
```diff
@@ -69,7 +70,10 @@
             appointmentService.cancelAppointment(id);
             return new ResponseEntity<>(HttpStatus.OK);
         } catch (Exception e) {
-            return new ResponseEntity<>(e.getMessage(), HttpStatus.BAD_REQUEST);
+            if (e instanceof TimeConstraintException) {
+                return new ResponseEntity<>("Cant cancel within 7 days.", HttpStatus.BAD_REQUEST);
```
Comment (sandyl289) [inline RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/controller/AppointmentController.java:74]:
```suggestion
                return new ResponseEntity<>("Can't cancel within 7 days.", HttpStatus.BAD_REQUEST);
```

**Review by sandyl289 — APPROVED** [review]
Really nice UI ✨ 💯

All tests passed (Repository & Service)

File: RepairShop-Frontend/src/components/ViewAppointments.vue [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+      idToDateTimeMap: {}
+    };
+  },
+
+  //fetch all of this customer's appointments and display them in a table
+  created: function() {
+    this.getAppointments();
+  },
+  methods: {
+    // Should output something like "Tue Mar 02 2021 10:00:00 GMT-0500 (Eastern Standard Time)" given a timestamp
```
Comment (T-Syme) [inline RepairShop-Frontend/src/components/ViewAppointments.vue:69]: Nice to see this function being reused! 😆

The comment above (line 69) corresponds to line 71 by the way. It was just a note to myself for the format for the slicing. 

The function changes a timestamp string (like 2021-03-02T15:00:00.000+00:00) to "Tue Mar 02, 2021 at 10:00" for my time zone

File: RepairShop-Frontend/src/components/ViewAppointments.vue [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+        .then(response => {
+          this.items = [];
+          this.getAppointments();
+          console.log(response);
+        })
+        .catch(error => {
+          if (error.response) {
+            if (error.response.status === 400) {
+              alert(
+                "You can only cancel appointments at least 7 days in  advance!"
```
Comment (T-Syme) [inline RepairShop-Frontend/src/components/ViewAppointments.vue:98]: You can probably just ignore this tiny detail, but there's an extra space
```suggestion
                "You can only cancel appointments at least 7 days in advance!"
```

**Review by T-Syme — APPROVED** [review]
Everything looks good!

I left a couple comments, but they can safely be ignored

**Review by Shaswata79 — APPROVED** [review]

**Pull request merged**